### PR TITLE
Move generated code

### PR DESCRIPTION
**Pull request by monopole**
To complete laying the groundwork for the `api` module, the generated builtin plugin code that the API depends on must move into the `api` module.

After this PR, the `kustomize/plugin` tree will contain no importable code.  Just `main` modules or random scripts.

### CODE REVIEW

File: travis/pre-commit.sh [hunk 1]
```diff
@@ -32,7 +53,11 @@ function runFunc {
 }
 
 function testGoLangCILint {
-  go run "github.com/golangci/golangci-lint/cmd/golangci-lint" run ./...
+  # TODO(2018/Oct/19): After the API is in place, reinstate this
```
Comment (Liujingfang1) [inline travis/pre-commit.sh]: The date, 2019?

File: travis/pre-commit.sh [hunk 2]
```diff
@@ -14,8 +14,29 @@ function installHelm {
...
 
+function removeBin {
+  local d=$(go env GOPATH)/bin/$1
+  echo "Removing binary $d"
+  /bin/rm -f $d
+}
+
 function installTools {
-  go install sigs.k8s.io/kustomize/pluginator
+  # TODO(2018/Oct/19): After the API is in place, and
```
Comment (Liujingfang1) [inline travis/pre-commit.sh]: The data, 2019?

Comment (monopole) [reply travis/pre-commit.sh]: augh!  will fix today!

File: travis/pre-commit.sh (hunk 1 above)
Comment (monopole) [reply travis/pre-commit.sh]: will fix!

**Comment by monopole** [comment]
Yes, it will - but wait a bit please!
Just a few lines.

**Pull request merged**